File: Java_Codes/array_binarySearch_recursive.java
import java.util.*;
public class array_binarySearch_recursive {

    int binarySearch(int arr[], int length, int y, int num) {
        while (y<=length) {
            int mid = y + (length-y)/2;

            if (arr[mid] == num) {
                return mid;
            }

            if (arr[mid]<num) {
                return binarySearch(arr, length, mid+1, num);
            }
            else{
                return binarySearch(arr, mid - 1, 0, num);
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int key;
        array_binarySearch_recursive func = new array_binarySearch_recursive();
        Scanner scr = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int size = scr.nextInt();

        int myArray[] = new int[size];
        System.out.println("Enter the elements of the Array: ");
        for (int i = 0; i < size; i++) {
            myArray[i] = scr.nextInt();
        }
        System.out.print("The entered array is: ");
        for (int i = 0; i < size; i++) {
            System.out.print(myArray[i]+" ");
        }

        System.out.print("\nEnter the element to be searched in the array: ");
        key = scr.nextInt();

        int result = func.binarySearch(myArray, size-1, 0, key);

        if (result==-1) {
            System.out.println("The element is not in the array.");
        }
        else{
            System.out.println("The element is at the "+result+"th indices of the array.");
        }
    scr.close();
    }
}

File: C++_Codes/array_binarySearch_recursive.cpp
#include <iostream>
int binarySearch(int arr[], int length, int y, int x)
{
    while (y <= length)
    {
        int mid = y + (length - y) / 2;
        if (arr[mid] == x)
        {
            return mid;
        }
        if (arr[mid] < x)
        {
            return binarySearch(arr, length, mid+1, x);
        }
        else
        {
            return binarySearch(arr, mid-1, 0, x);
        }
    }
    return -1;
}
int main()
{
    int size, key;

    std::cout << "Enter the size of the array: ";
    std::cin >> size;

    int myArray[size];
    printf("Enter the elements of the array: ");
    for (int i = 0; i < size; i++)
    {
        std::cin >> myArray[i];
    }
    std::cout << "The entered array is: ";
    for (int i = 0; i < size; i++)
    {
        printf("%d ", myArray[i]);
    }

    printf("\nEnter the element to be searched in the array: ");
    scanf("%d", &key);

    int result = binarySearch(myArray, size-1, 0, key);
    if (result == -1)
    {
        printf("The element is not in the array.");
    }
    else
    {
        printf("The element is at %dth indices of the array.", result);
    }
    return 0;
}